### PR TITLE
feat: refactor, and add possibility for multiple collectors on the same tenant

### PR DESCRIPTION
**Pull request by kristofgyuracz**
Signed-off-by: Kristof Gyuracz <kristof.gyuracz@axoflow.com>

### CODE REVIEW

File: internal/controller/telemetry/otel_conf_gen_test.go [hunk 1]
```diff
@@ -134,9 +132,16 @@ func TestOtelColConfComplex(t *testing.T) {
...
 	tenant := inputCfg.Tenants[0]
-	inputCfg.TenantSubscriptionMap[tenant.NamespacedName()] = getSubscriptionNamesFromSubscription(inputCfg.Subscriptions)
+
+	keysOfMap := func(subsMap map[v1alpha1.NamespacedName]v1alpha1.Subscription) (names []v1alpha1.NamespacedName) {
+		for key, _ := range subsMap {
+			names = append(names, key)
+		}
+		return
+	}
+	inputCfg.TenantSubscriptionMap[tenant.Name] = keysOfMap(inputCfg.Subscriptions)
```
Comment (OverOrion) [inline internal/controller/telemetry/otel_conf_gen_test.go:144]: [nit]
We could use `maps.Keys` here instead:
https://pkg.go.dev/golang.org/x/exp/maps#Keys

File: internal/controller/telemetry/route_controller.go [hunk 2]
```diff
@@ -0,0 +1,399 @@
...
+		for _, selectedSubscription := range selectedSubscriptions {
+			if subscriptionReferencing.Name == selectedSubscription.Name {
+				selected = true
+				break
+			}
+		}
+
+		if !selected {
+			subscriptionsToDisown = append(subscriptionsToDisown, subscriptionReferencing)
+		}
```
Comment (OverOrion) [inline internal/controller/telemetry/route_controller.go:350]: [nit]
I think this would be shorter with `slices.IndexFunc`, what do you think?

**Review by OverOrion — APPROVED** [review]
Thank you, this is a huge advancement :rocket: , overall looks good to me, I only had two minor remarks.

File: internal/controller/telemetry/otel_conf_gen_test.go (hunk 1 above)
Comment (kristofgyuracz) [reply internal/controller/telemetry/otel_conf_gen_test.go:144]: Checked @siliconbrain's [seqs](https://github.com/siliconbrain/go-mapseqs) library, but sadly seqs can't solve everything. :(

Comment (siliconbrain) [reply internal/controller/telemetry/otel_conf_gen_test.go:144]: there's always [mapseqs](https://github.com/siliconbrain/go-mapseqs) 😁

**Review by OverOrion — APPROVED** [review]

**Pull request merged**